packages/core/src/types.ts

```diff
@@ -571,55 +550,21 @@ export interface MachineConfig<
   /**
    * The initial context (extended state)
    */
-  context?: TContext | (() => TContext);
+  context?: TContext;
```

Comment:
Dropping support for `(() => TContext)` should also be noted down - even if it's temporary, because we can't be sure when we are going to bring it back.